server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputer.java

```diff
@@ -339,6 +361,20 @@ public DesiredBalance compute(
                     iterations
                 )
             );
+
+            if (assignedSomeNewlyCreatedShards
```

Comment:
This seemed to be the simplest fix, AFAICT. Similar to how we interrupt the computation when there is a new cluster state.

Reply:
I'm still wandering around the balancer code, but I saw [this bit of code](https://github.com/elastic/elasticsearch/blob/ca193bb923bf2a42df06dd39c8ad50068842879a/server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceShardsAllocator.java#L138-L145) in the ContinuousComputation instance that makes me wonder if a new computation task will be queued up, if we exit the code like this? I don't think I fully understand things yet, so I could be wrong.

Reply:
(w/o further digging into this, seems that returning a flag from the `compute` to signal we'd want to give the same input another compute round might be a solution).